src/core/Ekubo.sol

```diff
@@ -254,8 +224,8 @@ abstract contract Ekubo is SettlerAbstract {
                 // update data length
                 mstore(data, size)
 
-                // update free memory pointer (word-aligned)
-                mstore(0x40, add(data, and(add(add(size, 32), 31), not(31))))
+                // update free memory pointer
+                mstore(0x40, add(0x20, add(data, size)))
```

Comment:
Ahh, as a special case here, because `size := add(0x124, sig.length)`, it's actually faster/cheaper to write this as `add(data, add(0x20, size))` because the compiler will replace it with `add(data, add(0x144, sig.length))`